examples/window.rs

```diff
@@ -217,6 +244,29 @@ impl Application {
                 }
             },
             Action::RequestResize => window.swap_dimensions(),
+            Action::DumpMonitors => {
+                #[cfg(web_platform)]
+                {
+                    let future = event_loop.request_detailed_monitor_permission();
+                    let proxy = event_loop.create_proxy();
+                    let sender = self.sender.clone();
+                    wasm_bindgen_futures::spawn_local(async move {
+                        if let Err(error) = future.await {
+                            error!("{error}")
+                        }
+
+                        let _ = sender.send(Action::DumpMonitors);
+                        proxy.wake_up();
+                    });
```

Comment:
> What's the reason behind all of that and why web can not do what everyone else is doing here?

No other backend needs permission, so this is really impossible to do cross-platform the same cross-platform.
> I understand the permission part, but I don't get why we're spawning a future?

You need to know when the permission has been given. How else do you propose doing this?

Reply:
Generally I'd have an extension just for web based on trait that calls back to you once you got the permission. You can even do that for all permissions you want for whatever reason, not just monitor.

In the example, I'd probably just ask unconditionally during startup, though it could annoy or ask and just dumb anyway, since it'll be much simpler...

Reply:
> Generally I'd have an extension just for web based on trait that calls back to you once you got the permission. You can even do that for all permissions you want for whatever reason, not just monitor.

Uhm, you mean in the future, because right now that's not possible, right?

> In the example, I'd probably just ask unconditionally during startup, though it could annoy or ask and just dump anyway, since it'll be much simpler...

Asking for permission at startup isn't really an option, that would be an awful example.
But I can dump anyway, I think that's a good idea!

Reply:
Hm, on second that I find this actually useful as an example for how to let users deal with the async APIs on Web.
So I think removing this is a mistake, especially for the example.

I will split it up though, to make it easier to read.

Reply:
>Uhm, you mean in the future, because right now that's not possible, right?

yeah...